cmake/HalideGeneratorHelpers.cmake

```
@@ -493,6 +493,12 @@ function(add_halide_library TARGET)
         message(AUTHOR_WARNING "Arguments to add_halide_library were not recognized: ${ARG_UNPARSED_ARGUMENTS}")
     endif ()
 
+     if (NOT "${ARG_KEYWORDS_MISSING_VALUES}" STREQUAL "")
+        foreach(name IN LISTS ARG_KEYWORDS_MISSING_VALUES)
+            message(AUTHOR_WARNING "The argument ${name} passed to add_halide_library is undefined")
```

Our multi-valued arguments (except `TARGETS`) are allowed to be left empty.

```suggestion
            if (NOT name IN_LIST multiValueArgs OR name STREQUAL "TARGETS")
                message(AUTHOR_WARNING "The argument ${name} passed to add_halide_library is undefined")
            endif ()
```

I also wonder if `AUTHOR_WARNING` should be `SEND_ERROR`... I guess you get a valid build.